src/server/dfly_main.cc

```diff
@@ -691,25 +662,26 @@ Usage: dragonfly [FLAGS]
   if (memory.swap_total != 0)
     LOG(WARNING) << "SWAP is enabled. Consider disabling it when running Dragonfly.";
 
-  if (GetFlag(FLAGS_maxmemory).value == 0) {
+  dfly::max_memory_limit = dfly::GetMaxMemoryFlag();
```

Comment:
Where is this variable defined / declared? Did you mean to use a local var?

Reply:
no, it's declared in common.h and I meant to use it :)

Reply:
So there are 2 variables tracking memory size? What is the difference between them? Why do we need both? Which is the source of truth?
Should we make the flag only the initial value of `dfly::max_memory_limit` and ignore that flag after process startup?

Reply:
1. In this particular case I preferred to have a single global integer instead of going through flags. The var could be removed  but I consider it out of scope of this PR.
2. For most cases, I believe we will have a flag to be a single source of truth until we introduce extensions to the registry.

Reply:
I still need to pass the flag though because I need to parse values in a generic manner inside ConfigRegistry::Set

Reply:
Ok, sure. Sounds good. Consider adding a `// TODO` to remove the variable and only use the flag